server: provide a default for tgt_cmd_extra_args option

idryomov opened the pull request:
The default of "no extra args" is reflected in ceph-nvmeof.conf.  An example value is provided to demonstrate syntax.

trociny reviewed (approved):
I am ok with this though I am a bit concerned.

May be it is just my personal experience and there is no foundation for this, but when I see an original config file that comes from a distribution with many options commented I suppose the commented values are defaults. 

May be we should add some comments  to tell what the defaults are?

And I think much more options could use `get_with_default` and be commented in the config. I would leave only those that really cannot have a default value.

Though this is may be out of the scope of this PR.

idryomov commented:
> May be it is just my personal experience and there is no foundation for this, but when I see an original config file that comes from a distribution with many options commented I suppose the commented values are defaults.

Yeah, with `sshd_config` file being the primary example that everyone is used to.

> May be we should add some comments to tell what the defaults are?
> 
> And I think much more options could use `get_with_default` and be commented in the config. I would leave only those that really cannot have a default value.

+1

> 
> Though this is may be out of the scope of this PR.

This PR just follows the precedent of `transport_tcp_options` which clearly gives an example value (because ceph-nvmeof default there is an empty string and SPDK default for e.g. `max_io_size` is 128K, not 4M).  If we wanted to walk away from that, I would propose the following:
```
# Example value: -m 0x3 -L all
# tgt_cmd_extra_args =

# Example value: {"max_queue_depth" : 16, "max_io_size" : 4194304, "io_unit_size" : 1048576, "zcopy" : false}
# transport_tcp_options =
```

trociny commented:
```
# Example value: -m 0x3 -L all
# tgt_cmd_extra_args =

# Example value: {"max_queue_depth" : 16, "max_io_size" : 4194304, "io_unit_size" : 1048576, "zcopy" : false}
# transport_tcp_options =
```

LGTM!

idryomov committed:
server: provide a default for tgt_cmd_extra_args option

The default of "no extra args" is reflected in ceph-nvmeof.conf.
An example value is provided to demonstrate syntax.

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

idryomov committed:
ceph-nvmeof.conf: reflect transport_tcp_options default

Gateway's default for transport_tcp_options is "no extra options" and
we don't control SPDK's default for e.g. max_io_size.  Try to make that
clear as many people expect commented out values in an upstream-provided
config file to be defaults.

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

idryomov commented:
Rebased to address a trivial formatting conflict with #29.
